statistics/statistics.go

```diff
@@ -53,26 +55,31 @@ func Meta() *plugin.PluginMeta {
 		[]string{plugin.SnapGOBContentType})
 }
 
+// New() returns a new instance of this
 func New() *Plugin {
 	buffer := make(map[string][]interface{})
 	p := &Plugin{buffer: buffer,
 		bufferMaxSize: 100,
 		bufferCurSize: 0,
-		bufferIndex:   0}
+		bufferIndex:   0,
+		logger:        log.New()}
 	return p
 }
 
-func (p *Plugin) calculateStats(buff interface{}, logger *log.Logger) (map[string]float64, error) {
-	result := make(map[string]float64)
+// calculateStats calaculates the descriptive statistics for buff
+func (p *Plugin) calculateStats(buff interface{}) (map[string]interface{}, error) {
```

Comment:
using map[string][]float64 is better than map[string]interface as it only allows float64.